http/get_simple/python/server/server.py

```diff
@@ -56,24 +56,44 @@ def GetPutData():
     
     return batches
 
-def make_reader(schema, batches):
-    return pa.RecordBatchReader.from_batches(schema, batches)
-
-def generate_batches(schema, reader):
+def generate_buffers(schema, source):
     with io.BytesIO() as sink, pa.ipc.new_stream(sink, schema) as writer:
-        for batch in reader:
+        for batch in source:
             sink.seek(0)
-            sink.truncate(0)
             writer.write_batch(batch)
+            sink.truncate()
             yield sink.getvalue()
```

Comment:
Note that `getvalue` returns a copy of the internal BytesIO buffer. Instead you can use `getbuffer`:
https://docs.python.org/3/library/io.html#io.BytesIO.getbuffer

Reply:
If I use that I have to put a `del buffer` in the loop to avoid the `BufferError: Existing exports of data: object cannot be re-sized` error.

But a clearer way seems to be yielding the `BytesIO` object itself.

Reply:
Using a `with` statement would be more reliable then:
```python
    with sink.getbuffer() as buf:
        yield buf
```

Reply:
To avoid the `del buffer` which I think is kinda ugly and confusing, I ended up with this:

```python
        def write_chunk(buffer):
           if chunked:
               self.wfile.write('{:X}\r\n'.format(len(buffer)).encode('utf-8'))
           self.wfile.write(buffer)
           if chunked:
               self.wfile.write('\r\n'.encode('utf-8'))
           self.wfile.flush()
        foreach_batch_buffer(schema, source, write_chunk)
```

This means I can pass `sing.getbuffer()` to the `write_chunk` callback without having to `del` it when it goes out of scope. This also makes the version that chunks the buffer themselves avoid the need for more `del` statements.

Do you think this is acceptable or the `yield` solution with `del` statements is preferred?

Reply:
See my comment above. The `del` is unidiomatic is this case, while `with` triggers reliable closing of the buffer view.

Reply:
Sorry. Github didn't show me your comment before I posted mine 💢